Get rid of now broken fluidsynth_1

`fluidsynth_1` no longer compiles with GLib 2.70:

```
In file included from /nix/store/rrmzjd6gybl8ynnlnclxmla9rvcpw0cc-glib-2.70.0-dev/include/glib-2.0/glib/deprecated/gthread.h:124,
                 from /nix/store/rrmzjd6gybl8ynnlnclxmla9rvcpw0cc-glib-2.70.0-dev/include/glib-2.0/glib.h:112,
                 from /build/source/src/utils/fluidsynth_priv.h:25,
                 from /build/source/src/drivers/fluid_jack.c:31:
/build/source/src/drivers/fluid_jack.c:103:42: error: extra brace group at end of initializer
  103 | static fluid_mutex_t last_client_mutex = G_STATIC_MUTEX_INIT;     /* Probably not necessary, but just in case drivers are created by multiple threads */
      |                                          ^~~~~~~~~~~~~~~~~~~
/build/source/src/drivers/fluid_jack.c:103:42: note: (near initialization for 'last_client_mutex')
/build/source/src/drivers/fluid_jack.c:103:42: error: extra brace group at end of initializer
  103 | static fluid_mutex_t last_client_mutex = G_STATIC_MUTEX_INIT;     /* Probably not necessary, but just in case drivers are created by multiple threads */
      |                                          ^~~~~~~~~~~~~~~~~~~
/build/source/src/drivers/fluid_jack.c:103:42: note: (near initialization for 'last_client_mutex')
In file included from /nix/store/f1r94qx7a8m8iz7fxvlq655gbawql7xf-glibc-2.33-50-dev/include/bits/thread-shared-types.h:74,
                 from /nix/store/f1r94qx7a8m8iz7fxvlq655gbawql7xf-glibc-2.33-50-dev/include/bits/pthreadtypes.h:23,
                 from /nix/store/f1r94qx7a8m8iz7fxvlq655gbawql7xf-glibc-2.33-50-dev/include/sys/types.h:227,
                 from /nix/store/f1r94qx7a8m8iz7fxvlq655gbawql7xf-glibc-2.33-50-dev/include/stdlib.h:394,
                 from /nix/store/rrmzjd6gybl8ynnlnclxmla9rvcpw0cc-glib-2.70.0-dev/include/glib-2.0/glib/gutils.h:433,
                 from /nix/store/rrmzjd6gybl8ynnlnclxmla9rvcpw0cc-glib-2.70.0-dev/include/glib-2.0/glib/gthread.h:34,
                 from /nix/store/rrmzjd6gybl8ynnlnclxmla9rvcpw0cc-glib-2.70.0-dev/include/glib-2.0/glib/gasyncqueue.h:32,
                 from /nix/store/rrmzjd6gybl8ynnlnclxmla9rvcpw0cc-glib-2.70.0-dev/include/glib-2.0/glib.h:32,
                 from /build/source/src/utils/fluidsynth_priv.h:25,
                 from /build/source/src/drivers/fluid_jack.c:31:
/build/source/src/drivers/fluid_jack.c:103:42: error: extra brace group at end of initializer
  103 | static fluid_mutex_t last_client_mutex = G_STATIC_MUTEX_INIT;     /* Probably not necessary, but just in case drivers are created by multiple threads */
      |                                          ^~~~~~~~~~~~~~~~~~~
/build/source/src/drivers/fluid_jack.c:103:42: note: (near initialization for 'last_client_mutex')
/build/source/src/drivers/fluid_jack.c:103:42: warning: excess elements in union initializer
  103 | static fluid_mutex_t last_client_mutex = G_STATIC_MUTEX_INIT;     /* Probably not necessary, but just in case drivers are created by multiple threads */
      |                                          ^~~~~~~~~~~~~~~~~~~
/build/source/src/drivers/fluid_jack.c:103:42: note: (near initialization for 'last_client_mutex')
/build/source/src/drivers/fluid_jack.c:103:13: warning: Deprecated pre-processor symbol: replace with "g_mutex_init"
  103 | static fluid_mutex_t last_client_mutex = G_STATIC_MUTEX_INIT;     /* Probably not necessary, but just in case drivers are created by multiple threads */
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/libfluidsynth.dir/build.make:90: src/CMakeFiles/libfluidsynth.dir/drivers/fluid_jack.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:258: src/CMakeFiles/libfluidsynth.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

The failure looks patchable: We'd need to replace the use of `G_STATIC_MUTEX_INIT` in `fluid_jack.c` by an equivalent call to `g_mutex_init`. However I'm not quite confident about the right place to call it, since we need to avoid calling it twice (which is UB according to the GLib documentation) and it looks like there are two different intialization routines that'd need to initialize the lock, `new_fluid_jack_client ` and `new_fluid_jack_midi_driver`.

Therefore I think should try to make an effort to finally get rid of this vulnerable version of fluidsynth (#124624). Remaining uses:

* [x] `csound`, fixed in #141506
* [x] `lmms` (cc @cillianderoiste, @yegortimoshenko), fixed in #141512
* [x] `gzdoom` (cc @Lassulus), fixed in #141517

--- a/nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
+++ b/nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
@@ -1160,6 +1160,14 @@ Superuser created successfully.
           will no longer work and must be updated.
         </para>
       </listitem>
+      <listitem>
+        <para>
+          The <literal>fluidsynth_1</literal> attribute has been
+          removed, as this legacy version is no longer needed in
+          nixpkgs. The actively maintained 2.x series is available as
+          <literal>fluidsynth</literal> unchanged.
+        </para>
+      </listitem>
     </itemizedlist>
   </section>
   <section xml:id="sec-release-21.11-notable-changes">

--- a/nixos/doc/manual/release-notes/rl-2111.section.md
+++ b/nixos/doc/manual/release-notes/rl-2111.section.md
@@ -358,6 +358,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 - The `services.mosquitto` module has been rewritten to support multiple listeners and per-listener configuration.
   Module configurations from previous releases will no longer work and must be updated.
 
+- The `fluidsynth_1` attribute has been removed, as this legacy version is no longer needed in nixpkgs. The actively maintained 2.x series is available as `fluidsynth` unchanged.
+
 ## Other Notable Changes {#sec-release-21.11-notable-changes}
 
 

--- a/pkgs/applications/audio/fluidsynth/default.nix
+++ b/pkgs/applications/audio/fluidsynth/default.nix
@@ -1,33 +1,17 @@
 { stdenv, lib, fetchFromGitHub, pkg-config, cmake
 , alsa-lib, glib, libjack2, libsndfile, libpulseaudio
 , AudioUnit, CoreAudio, CoreMIDI, CoreServices
-, version ? "2"
 }:
 
-let
-  versionMap = {
-    "1" = {
-      fluidsynthVersion = "1.1.11";
-      sha256 = "0n75jq3xgq46hfmjkaaxz3gic77shs4fzajq40c8gk043i84xbdh";
-    };
-    "2" = {
-      fluidsynthVersion = "2.2.3";
-      sha256 = "0x5808d03ym23np17nl8gfbkx3c4y3d7jyyr2222wn2prswbb6x3";
-    };
-  };
-in
-
-with versionMap.${version};
-
-stdenv.mkDerivation  {
-  name = "fluidsynth-${fluidsynthVersion}";
-  version = fluidsynthVersion;
+stdenv.mkDerivation rec {
+  pname = "fluidsynth";
+  version = "2.2.3";
 
   src = fetchFromGitHub {
     owner = "FluidSynth";
     repo = "fluidsynth";
-    rev = "v${fluidsynthVersion}";
-    inherit sha256;
+    rev = "v${version}";
+    sha256 = "0x5808d03ym23np17nl8gfbkx3c4y3d7jyyr2222wn2prswbb6x3";
   };
 
   nativeBuildInputs = [ pkg-config cmake ];

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -24638,7 +24638,6 @@ with pkgs;
   fluidsynth = callPackage ../applications/audio/fluidsynth {
     inherit (darwin.apple_sdk.frameworks) AudioUnit CoreAudio CoreMIDI CoreServices;
   };
-  fluidsynth_1 = fluidsynth.override { version = "1"; };
 
   fmit = libsForQt5.callPackage ../applications/audio/fmit { };
 

